tests/trace/test_image_thread_safety.py

```diff
@@ -0,0 +1,56 @@
+import os
+import random
+from concurrent.futures import ThreadPoolExecutor
+
+import PIL
+import pytest
+
+
+@pytest.fixture()
+def deferred_cleanup():
+    files_to_cleanup = []
+
+    def add_file(name: str):
+        files_to_cleanup.append(name)
+
+    def cleanup():
+        for name in files_to_cleanup:
+            os.remove(name)
+
+    try:
+        yield add_file
+    finally:
+        cleanup()
```

Comment:
You can use TemporaryFile for this:
https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
ack